http/src/main/java/io/micronaut/http/reactive/execution/ReactorExecutionFlowImpl.java

```diff
@@ -111,6 +112,13 @@ public void onComplete() {
         });
     }
 
+    @Nullable
+    @Override
+    public ImperativeExecutionFlow<Object> asDone() {
```

Comment:
I wonder if we can also detect resolved Mono/Flux

Reply:
i don't think so, we can't subscribe without the potential for side effects, and i don't see any api to check specifically whether the mono is side-effect-free

Reply:
ive looked in reactor and found one approach, i dont think it gets better unfortunately.